src/rules/no-restricted-patterns.ts

```diff
@@ -40,34 +49,44 @@ export function run({feature}: GherkinData, configuration: Configuration): RuleE
     checkNameAndDescription(node, restrictedPatterns, language, errors);
 
     // And all the steps of each child
-    node.steps.forEach(step => {
+    node.steps.forEach((step, index) => {
+      checkStepNode(step, node.steps[index], restrictedPatterns, language, errors);
       checkStepNode(step, node, restrictedPatterns, language, errors);
+
     });
   });
-
-  return errors;
+  return errors.filter((obj, index, self) =>
+    index === self.findIndex((el) => el.message === obj.message)
+  );
 }
 
 function getRestrictedPatterns(configuration: Configuration): ConfigurationPatterns {
   // Patterns applied to everything; feature, scenarios, etc.
   const globalPatterns = (configuration.Global || []).map(pattern => new RegExp(pattern, 'i'));
-
+  //pattern to apply on all steps
+  const stepPatterns = (configuration.Step || []).map(pattern => new RegExp(pattern, 'i'));
   const restrictedPatterns = {} as ConfigurationPatterns;
+
   Object.keys(availableConfigs).forEach((key: keyof Configuration) => {
     const resolvedKey = key.toLowerCase().replace(/ /g, '') as keyof Configuration;
     const resolvedConfig = (configuration[key] || []);
-
     restrictedPatterns[resolvedKey] = resolvedConfig.map(pattern => new RegExp(pattern, 'i'));
-
+    if (keywords.map(item => item.toLowerCase()).includes(resolvedKey.toLowerCase())) {
+      restrictedPatterns[resolvedKey] = restrictedPatterns[resolvedKey].concat(stepPatterns);
+    }
     restrictedPatterns[resolvedKey] = restrictedPatterns[resolvedKey].concat(globalPatterns);
   });
-
   return restrictedPatterns;
 }
 
 function getRestrictedPatternsForNode(node: GherkinNode, restrictedPatterns: ConfigurationPatterns, language: string): RegExp[] {
   const key = gherkinUtils.getLanguageInsensitiveKeyword(node, language).toLowerCase() as keyof ConfigurationPatterns;
-
+  if (keywords.map(item => item.toLowerCase()).includes(key.toLowerCase())) {
+    previousKeyword = key;
+  }
+  if (key.toLowerCase() === 'and' && keywords.map(item => item.toLowerCase()).includes(previousKeyword.toLowerCase())) {
```

Comment:
This only affects to 'And' keywords, but has no effect for 'or' or "*"

Reply:
Hello, I had the problem that I mentioned above, about the library, is it solved? In order to review it, I can't run the tests .